cli/remotebazel/remotebazel.go

```diff
@@ -928,6 +928,21 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	for {
 		inRsp, executeResponse, latestErr = attemptRun(ctx, bbClient, execClient, req)
 
+		// Handle known error conditions.
+		if latestErr != nil {
+			if strings.Contains(latestErr.Error(), "failed to get snapshot") {
+				log.Warnf("The requested snapshot was not found. Failing the request...")
+				return 1, nil
+			} else if strings.Contains(latestErr.Error(), "requested executor ID not found") {
```

Comment:
Using string matching for errors tends to be brittle. Are we able to use Details here and match against a fixed error code? (see [AnonymousUserError](https://github.com/buildbuddy-io/buildbuddy/blob/cae17cae25fa766261f11941862c0464950fac05/server/util/authutil/authutil.go#L101-L110) for an example)

Reply:
I applied that approach for SnapshotNotFoundError. It doesn't work with RequestedExecutorNotFoundError because the [execution server doesn't forward on the details ](https://github.com/buildbuddy-io/buildbuddy/blob/master/enterprise/server/remote_execution/execution_server/execution_server.go#L701)